discord/http.py

```diff
@@ -1208,13 +1209,37 @@ def start_forum_thread(
 
         if rate_limit_per_user:
             payload["rate_limit_per_user"] = rate_limit_per_user
+
+        form = [{"name": "payload_json"}]
+        if files:
+            attachments = []
+            for index, file in enumerate(files):
+                attachments.append(
+                    {
+                        "id": index,
+                        "filename": file.filename,
+                        "description": file.description,
+                    }
+                )
+                form.append(
+                    {
+                        "name": f"files[{index}]",
+                        "value": file.fp,
+                        "filename": file.filename,
+                        "content_type": "application/octet-stream",
+                    }
+                )
+            payload["attachments"] = attachments
+
+        form[0]["value"] = utils._to_json(payload)
+
         # TODO: Once supported by API, remove has_message=true query parameter
         route = Route(
             "POST",
             "/channels/{channel_id}/threads?has_message=true",
             channel_id=channel_id,
         )
-        return self.request(route, json=payload, reason=reason)
+        return self.request(route, form=form, reason=reason)
```

Comment:
I can patch this so it still uses the old approach when there are no files, but this _does_ seem to work when there are no files.

I don't know what is going on behind this call, because I am new to this project - is calling this with the json kwarg simply more efficient than this?

Reply:
> I can patch this so it still uses the old approach when there are no files, but this did still seem to work in that case during my testing.
> 
> I don't know what is going on behind this call, because I am new to this project - is calling this with the json kwarg simply more efficient than this?

I'm not actually sure if it's more efficient, but since we only need to send JSON then we should probably just send JSON
A simple enough solution would be to check if `files` is empty after `rate_limit_per_user`

Reply:
what plun said

Reply:
Sorry for the delay in making this change.